packages/adminrouter/extra/src/includes/metrics-location-level.conf

```diff
@@ -0,0 +1,40 @@
+# Output fine-grained location directive level metrics using the VTS module.
+#
+# Caution:
+# A request which matches a location that has a VTS module filter defined will be
+# counted on the Nginx location directive level filter but it will not be counted
+# on a filter defined on the server directive level that it matches as well.
+#
+# The following filter expects a $upstream_tag to be set before importing in order to
+# identify the upstream that the metrics belong to. This constitutes a mechanism to
+# annotate the Admin Router config, allowing one to collect metrics of dynamically
+# resolved upstreams on the fly. It is also the name that will show up in Grafana when
+# referring to a particular upstream. Grouping metrics based on upstream would normally
+# only be possible with Nginx Plus because only then they can be defined in a upstream
+# directive and continue to be dynamically resolved.
+# https://tenzer.dk/nginx-with-dynamic-upstreams/
+
+# Create a counter for every backend that requests go to when passing this line in a location directive.
+# Tag the counter with the manually set upstream.
```

Comment:
❓ Add: This means that before including this file, a line like `set $upstream_tag <CUSTOM_UPSTREAM_TAG>;`.